### PR TITLE
Add runtime check for kotlin-native mt coroutines

### PR DESCRIPTION
**Pull request by e5l**
fixes #KTOR-956

### CODE REVIEW

File: ktor-client/ktor-client-core/posix/src/io/ktor/client/utils/CoroutineUtilsPosix.kt [hunk 1]
```diff
@@ -15,3 +16,16 @@ public actual fun Dispatchers.clientDispatcher(
...
     dispatcherName: String
 ): CoroutineDispatcher = Unconfined
+
+internal actual fun checkCoroutinesVersion() {
+    try {
+        val parent = Job()
+        parent.freeze()
+        Job(parent)
+    } catch (cause: Throwable) {
+        val message = "Invalid coroutines version. " +
```
Comment (cy6erGn0m) [inline ktor-client/ktor-client-core/posix/src/io/ktor/client/utils/CoroutineUtilsPosix.kt:26]: The word "invalid" is not perfect here. I would say something like "Native ktor HttpClient requires  kotlinx.coroutines version with ....." or "kotlinx.coroutines version with .... is required for ktor native"

File: ktor-client/ktor-client-core/posix/src/io/ktor/client/utils/CoroutineUtilsPosix.kt [hunk 2]
```diff
@@ -15,3 +16,16 @@ public actual fun Dispatchers.clientDispatcher(
...
+            "Consider using version with suffix `-native-mt`, like `1.3.9-native-mt` for native."
+
+        error(message)
```
Comment (cy6erGn0m) [inline ktor-client/ktor-client-core/posix/src/io/ktor/client/utils/CoroutineUtilsPosix.kt:29]: Shouldn't we use some specific exception, error or even crash?

**Review by cy6erGn0m — CHANGES_REQUESTED** [review]

File: ktor-client/ktor-client-core/posix/src/io/ktor/client/utils/CoroutineUtilsPosix.kt (hunk 1 above)
Comment (hhariri) [reply ktor-client/ktor-client-core/posix/src/io/ktor/client/utils/CoroutineUtilsPosix.kt:26]: +1. I think the more help we provide folks, the better

Comment (e5l) [reply ktor-client/ktor-client-core/posix/src/io/ktor/client/utils/CoroutineUtilsPosix.kt:26]: Thanks, will fix

File: ktor-client/ktor-client-core/posix/src/io/ktor/client/utils/CoroutineUtilsPosix.kt (hunk 2 above)
Comment (e5l) [reply ktor-client/ktor-client-core/posix/src/io/ktor/client/utils/CoroutineUtilsPosix.kt:29]: I'm not sure the purpose of the `specific` exception here: what is the reason for catching it?

File: ktor-client/ktor-client-core/posix/src/io/ktor/client/utils/CoroutineUtilsPosix.kt [hunk 3]
```diff
@@ -15,3 +16,16 @@ public actual fun Dispatchers.clientDispatcher(
...
 ): CoroutineDispatcher = Unconfined
+
+internal actual fun checkCoroutinesVersion() {
+    try {
+        val parent = Job()
+        parent.freeze()
+        Job(parent)
+    } catch (cause: Throwable) {
+        val message = "Ktor native HttpClient requires kotlinx.coroutines version with `native-mt` suffix" +
+            "(like `1.3.9-native-mt`). Consider checking the dependencies."
```
Comment (cy6erGn0m) [inline ktor-client/ktor-client-core/posix/src/io/ktor/client/utils/CoroutineUtilsPosix.kt:27]: space between word "suffix" and opening "("

**Review by cy6erGn0m — APPROVED** [review]

File: ktor-client/ktor-client-core/posix/src/io/ktor/client/utils/CoroutineUtilsPosix.kt (hunk 3 above)
Comment (hhariri) [reply ktor-client/ktor-client-core/posix/src/io/ktor/client/utils/CoroutineUtilsPosix.kt:27]: LGTM

**Review by hhariri — APPROVED** [review]

**Pull request merged**